### PR TITLE
Break on Property with long value and Literal key

### PR DESCRIPTION
**Pull request by azz**
Fixes an instability issue:

First print:

```patch
 const x = {
-  "ABC": "12345678901234567890123456789012345678901234567890123456789012345678901234567890"
+  ABC: "12345678901234567890123456789012345678901234567890123456789012345678901234567890"
 };
```

Second print:

```patch
 const x = {
-  ABC: "12345678901234567890123456789012345678901234567890123456789012345678901234567890"
+  ABC:
+    "12345678901234567890123456789012345678901234567890123456789012345678901234567890"
 };
```

Fix is to treat the `"ABC"` the same, and break if the value is too long.

This fixes the only remaining non-comment-related `prettier(input) !== prettier(prettier(input))` issue in #1643.

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -4173,7 +4173,9 @@ function printAssignment(
 
   const canBreak =
     (isBinaryish(rightNode) && !shouldInlineLogicalExpression(rightNode)) ||
-    ((leftNode.type === "Identifier" || leftNode.type === "MemberExpression") &&
+    ((leftNode.type === "Identifier" ||
+      leftNode.type === "Literal" ||
```
Comment (vjeux) [inline src/printer.js]: Literal only works for flow. You need to use isLiteral if you want it to work on babylon

Comment (vjeux) [inline src/printer.js]: Or check for ("StringLiteral" || "Literal" && typeof value === "string") like two line below.

**Comment by azz** [comment]
Was busy today, sorry for the delay.

Fix now works on babylon too. Did a bit of a clean up of existing string literal checks too.

**Comment by vjeux** [comment]
Awesome thanks!

**Pull request merged**